power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/meta_data.hpp

```diff
@@ -51,6 +51,24 @@ template <class T>
 struct ctype_t<T> : ctype_t<std::underlying_type_t<T>> {};
 template <class T> constexpr CType ctype_v = ctype_t<T>::value;
 
+// function selector based on ctype
+// the operator() of the functor should have a single template parameter
+// the selector will instantiate the operator() with relevant type
+template <class Functor, class... Args> decltype(auto) ctype_func_selector(CType ctype, Functor f, Args&&... args) {
```

Comment:
should we add `requires` clauses to enforce this behaviour? I know `requires` clauses for functors are often more difficult than for 'regular' template parameters, so i'm also fine with keeping it as is

Reply:
Let's keep it as is.